sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/application-changes-report-integration/_tests_/application-changes-report-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,383 @@
+            offering: newOffering,
+          },
+          {
+            initialValue: {
+              triggerType,
+              reportedDate: isReported ? new Date() : null,
+            },
+          },
+        ),
+      );
+      application.currentAssessment = newAssessment;
+      return db.application.save(application);
+    }
+    /**
+     * Create a new offering with different study dates for application changes.
+     * @param originalOffering original offering.
+     * @param offeringStartDateDifference difference in days
+     * from original offering start date.
+     * @param offeringEndDateDifference difference in days
+     * from original offering end date.
+     * @returns new offering.
+     */
+    async function createNewOfferingWithDifferentStudyDates(
+      originalOffering: EducationProgramOffering,
+      offeringStartDateDifference: number,
+      offeringEndDateDifference: number,
+    ): Promise<EducationProgramOffering> {
+      const newOfferingStartDate = getISODateOnlyString(
+        addDays(offeringStartDateDifference, originalOffering.studyStartDate),
+      );
+      const newOfferingEndDate = getISODateOnlyString(
+        addDays(offeringEndDateDifference, originalOffering.studyEndDate),
+      );
+      return db.educationProgramOffering.save(
+        createFakeEducationProgramOffering(
+          {
+            auditUser,
+            institutionLocation: originalOffering.institutionLocation,
+          },
+          {
+            initialValues: {
+              studyStartDate: newOfferingStartDate,
+              studyEndDate: newOfferingEndDate,
+              offeringIntensity: OfferingIntensity.partTime,
+            },
+          },
+        ),
+      );
+    }
+    /**
+     * Get application changes CSV file record.
+     * @param application application.
+     * @param loanType loan type.
+     * @param activity activity.
+     * @returns file record.
+     */
+    function getExpectedApplicationChangesCSVRecord(
+      application: Application,
+      loanType: "FT" | "PT",
+      activity: "Early Withdrawal" | "Reassessment",
+    ): string {
+      const sin = application.student.sinValidation.sin;
+      const user = application.student.user;
+      const newAssessmentDate = getPSTPDTDateTime(
+        application.currentAssessment.createdAt,
+        { dateTimeFormat: APPLICATION_CHANGES_DATE_TIME_FORMAT },
+      );
+      const originalOffering =
+        application.currentAssessment.previousDateChangedReportedAssessment
+          .offering;
+      const newOffering = application.currentAssessment.offering;
+      const institutionCode = newOffering.institutionLocation.institutionCode;
+      return `${sin},${user.firstName},${user.lastName},${application.applicationNumber},${loanType},${institutionCode},${originalOffering.studyStartDate},${originalOffering.studyEndDate},${activity},${newAssessmentDate},${newOffering.studyStartDate},${newOffering.studyEndDate}`;
```

Comment:
As a suggestion it can be returned also as below.
```ts
return [
        sin,
        user.firstName,
        user.lastName,
        application.applicationNumber,
        loanType,
        institutionCode,
        originalOffering.studyStartDate,
        originalOffering.studyEndDate,
        activity,
        newAssessmentDate,
        newOffering.studyStartDate,
        newOffering.studyEndDate,
      ].join(",");
```